### PR TITLE
Add console.warn for object mutation.

### PR DESCRIPTION
**Pull request by jdalton**

### CODE REVIEW

File: karma.config.js [hunk 1]
```diff
@@ -5,8 +5,9 @@
 const globby = require('globby');
 const istanbul = require('rollup-plugin-istanbul');
 const { nodeResolve } = require('@rollup/plugin-node-resolve');
-
+// eslint-disable-next-line import/newline-after-import
 process.env.CHROME_BIN = require('puppeteer').executablePath();
+process.env.NODE_ENV = 'test';
```
Comment (garychangsf) [inline karma.config.js:10]: where is this being used?

File: packages/near-membrane-base/src/membrane.ts [hunk 2]
```diff
@@ -1073,55 +828,62 @@ export function createMembraneMarshall() {
...
+                // Future optimization: Hoping proxies with frozen handlers can be faster.
                 ObjectFreeze(this);
             }
 
             private makeProxyStatic(shadowTarget: ShadowTarget) {
-                // assert: trapMutations must be true
                 const { foreignTargetPointer } = this;
                 const targetIntegrityTraits =
                     foreignCallableGetTargetIntegrityTraits(foreignTargetPointer);
+                if (DEV_MODE) {
```
[the remark this answers is not in the document]
Comment (jdalton) [reply packages/near-membrane-base/src/membrane.ts:848]: The `DEV_MODE` bit is compiled away and removed during the rollup bundle process.

File: karma.config.js (hunk 1 above)
Comment (jdalton) [reply karma.config.js:10]: This makes the environment var consistent across jest and karma. I was seeing karma use dev builds.

File: packages/near-membrane-base/src/membrane.ts [hunk 3]
```diff
@@ -1073,55 +828,62 @@ export function createMembraneMarshall() {
...
+                    try {
+                        foreignCallableWarn(
+                            'Mutations tracked in a separate object graph for object:',
```
[the remark this answers is not in the document]
Comment (garychangsf) [reply packages/near-membrane-base/src/membrane.ts:851]: I am thinking from developer's perspective. The shorter version you have does not make much sense for them.

File: packages/near-membrane-base/src/membrane.ts [hunk 4]
```diff
@@ -1088,6 +1091,17 @@ export function createMembraneMarshall() {
             private makeProxyStatic(shadowTarget: ShadowTarget) {
                 // assert: trapMutations must be true
                 const { foreignTargetPointer } = this;
+                if (DEV_MODE) {
+                    try {
+                        foreignCallableWarn(
+                            'Mutations on the membrane of an object originating ' +
```
Comment (garychangsf) [inline packages/near-membrane-base/src/membrane.ts:1097]: ❤️

**Review by garychangsf — APPROVED** [review]
Thanks for cleaning up the changes! 😌

File: packages/near-membrane-base/src/membrane.ts [hunk 5]
```diff
@@ -1088,6 +1091,17 @@ export function createMembraneMarshall() {
             private makeProxyStatic(shadowTarget: ShadowTarget) {
                 // assert: trapMutations must be true
                 const { foreignTargetPointer } = this;
+                if (DEV_MODE) {
+                    try {
+                        foreignCallableWarn(
```
Comment (manuel-jasso) [inline packages/near-membrane-base/src/membrane.ts:1096]: When does this happen? Would a user see thousands of these?

Comment (jdalton) [reply packages/near-membrane-base/src/membrane.ts:1096]: This happens only on the first set of a membraned object when the proxy flips to static or live. This built in dev-mode so not the default build. Still TBD on now to have LWC/LWR build near-membrane. For their builds LWC uses 'developer' and LWR uses 'dev' so there's still some normalizing to do.

**Pull request merged**